blog/_posts/2018-03-29-missing.md

```diff
@@ -0,0 +1,414 @@
+
+Other temporary limitations concern compiler optimizations which are not yet implemented.
+First, code involving missing values is [not yet](https://github.com/JuliaLang/julia/issues/23338)
+as efficient as it could be. Second, conversion between `Array{T}` and
+`Array{Union{Missing,T}}` currently involves a copy. In theory it should be possible
+to use the same memory for bits types, since only the type tag array differs.
+Third, the Julia compiler is [currently unable](https://github.com/JuliaLang/julia/issues/23336)
+to generate code using [SIMD](https://en.wikipedia.org/wiki/SIMD) instructions when `missing`
+values are involved. This is not insurmountable since masked SIMD instructions allow applying
+an operation only to some values (the non-missing ones). While the absence of SIMD reduces
+noticeably the performance of many operations, it appears that Julia already achieves
+the same speed as vectorized operations in R (which are implemented in C). So there is
+room for improvement, but the current situation is already reasonably efficient.
+
+A more fundamental limitation is inherent to the choice of the `Union{Missing,T}`
+representation. In this representation, a non-missing value does not carry any information
+about whether it could have been missing, i.e. about whether it has been extracted from an
+array or from a column of a data set which allows for missing values. In practice, this means
+that, if `x` is an `Array{Union{Missing,T}}` but does not actually contain missing values,
+`map(identity, x)` will return an `Array{T}`. This is because `map` chooses its return type
+based only on the actual contents of the output, to avoid depending on type inference
+(which can vary depending on compiler improvements). This also means that when applying
+a function to each element in an `Array{Union{T,Missing}}`, one cannot choose the result
+type based on the type of the first element, which can be problematic e.g. to decide
+whether a table column should allow for `NULL` entries in a SQL database. This issue has
+been discussed [at length](https://discourse.julialang.org/t/missing-data-and-namedtuple-compatibility/8136/)
+in several occasions, but it is not clear yet which mitigating approach is the best one.
+
+Despite these limitations, we believe that missing values support in Julia 0.7 will be one
+of the most complete even among specialized statistical languages.
+
+**Author**: [Milan Bouchet-Valat](http://bouchet-valat.site.ined.fr), Research scientist,
+French National Institute for Demographic Studies (Ined), Paris.
+
+**Acknowledgements**: This framework is the result of collective efforts over several
+years. John Myles White lead the reflection around missing values support in Julia
+until 2016. Jameson Nash implemented compiler optimizations, and Jacob Quinn
+implemented the efficient memory layout for arrays. Alex Arslan, Jeff Bezanson,
+Stefan Karpinski, Jameson Nash and Jacob Quinn have been the most central
```

Comment:
you can include yourself in this list as well :) I think @davidanthoff would be a good mention as well, with lots of contributions on things to consider w/ missing values.

Reply:
As the author of the post, I'm not sure I also need to be in the list. ;-)

I'll add David too.